tests/v2/validation/rbac/etcdbackup/etcd_backup_test.go

```diff
@@ -0,0 +1,115 @@
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type ETCDRbacBackupTestSuite struct {
+	suite.Suite
+	client  *rancher.Client
+	session *session.Session
+	cluster *management.Cluster
+}
+
+func (rb *ETCDRbacBackupTestSuite) TearDownSuite() {
+	rb.session.Cleanup()
+}
+
+func (rb *ETCDRbacBackupTestSuite) SetupSuite() {
+	rb.session = session.NewSession()
+
+	client, err := rancher.NewClient("", rb.session)
+	require.NoError(rb.T(), err)
+
+	rb.client = client
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(rb.T(), clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(rb.client, clusterName)
+	require.NoError(rb.T(), err, "Error getting cluster ID")
+	rb.cluster, err = rb.client.Management.Cluster.ByID(clusterID)
+	require.NoError(rb.T(), err)
+}
+
+func (rb *ETCDRbacBackupTestSuite) testEtcdSnapshotCluster(role, name string, user *management.User) {
+
+	rb.T().Logf("Created user: %v", user.Username)
+	standardUserClient, err := rb.client.AsUser(user)
+	require.NoError(rb.T(), err)
+
+	adminProject, err := rbac.CreateProject(rb.client, rb.cluster.ID)
+	require.NoError(rb.T(), err)
+
+	if role == rbac.StandardUser {
+		if strings.Contains(role, "project") {
+			err := users.AddProjectMember(rb.client, adminProject, user, role, nil)
+			require.NoError(rb.T(), err)
+		} else {
+			err := users.AddClusterRoleToUser(rb.client, rb.cluster, user, role, nil)
+			require.NoError(rb.T(), err)
+		}
+	}
+
+	relogin, err := standardUserClient.ReLogin()
+	require.NoError(rb.T(), err)
+	standardUserClient = relogin
+
+	log.Info("Test case - Take Etcd snapshot of a cluster as a " + name)
+
+	err = etcdsnapshot.CreateRKE2K3SSnapshot(standardUserClient, rb.cluster.ID)
+	switch role {
+	case rbac.RoleClusterOwner, rbac.RestrictedAdmin:
+		require.NoError(rb.T(), err)
+
+	case rbac.RoleClusterMember, rbac.RoleProjectOwner, rbac.RoleProjectMember:
+		require.Error(rb.T(), err)
+		assert.Equal(rb.T(), "Resource type [provisioning.cattle.io.cluster] is not updatable", err.Error())
+	}
+}
+
+func (rb *ETCDRbacBackupTestSuite) TestETCDRBAC() {
+	tests := []struct {
+		name   string
+		role   string
+		member string
+	}{
+		{"Cluster Owner", rbac.RoleClusterOwner, rbac.StandardUser},
+		{"Cluster Member", rbac.RoleClusterMember, rbac.StandardUser},
+		{"Project Owner", rbac.RoleProjectOwner, rbac.StandardUser},
+		{"Project Member", rbac.RoleProjectMember, rbac.StandardUser},
+		{"Restricted Admin", rbac.RestrictedAdmin, rbac.RestrictedAdmin},
+	}
+	for _, tt := range tests {
+		clusterID, err := clusters.GetClusterIDByName(rb.client, rb.cluster.ID)
+		require.NoError(rb.T(), err)
+		if !(strings.Contains(clusterID, "c-m-")) {
+			rb.T().Skip("Skipping tests since cluster is not of type - k3s or RKE2")
```

Comment:
besides this line, how will the user running the test know this is only for rke2/k3s? Can you make sure to have a clear readme, and rename the suite to reflect this?

Reply:
Sure, can update the Readme.